modules/script/script-impl/README.md

```diff
@@ -0,0 +1,79 @@
+# Enonic XP global TS types
```

Comment:
script-impl is an implementation detail module.
You probably meant to put it into script-api

Reply:
Not really. This definitions, that will actually be called `base`, are not tightly coupled with some specific module. The reason why I put it there was  actually because of the implementation, for the developer after changing or adding something like [this pre-init code](https://github.com/enonic/xp/blob/5c6b85b194035f2b2e21f036b8909baa8747460e/modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/executor/ScriptExecutorImpl.java#L40) shouldn't forget to add types for it.

Reply:
Maybe it should be called `common`, not `base`?

Reply:
I don't like the name `common` because we don't provide some common types there, that can be used across apps. We provide definitions for the core properties of the global object: `__`, `log`, etc. I would not name it `core`, because the name implies, that we need it to use with other types, which is not actually true.

Reply:
`global`, `base`, `common`, `core`, `main`, `general` are actually names, that I was thinking about. They are abstract, each and every of them have their downside. But any of them will do anyway.

Reply:
`global` is good.
@rymsha any objections?

Reply:
This is a tough one... but "log, require, resolve, __, exports, module" are still part of our script-api. I'll have to thing about it more...

Reply:
Will move code to `lib/typescript`. The package will be named `global`.